src/Data/RFC5322.hs

```diff
@@ -233,6 +234,9 @@ special = satisfy isSpecial
 rfc5422DateTimeFormat :: String
 rfc5422DateTimeFormat = "%a, %d %b %Y %T %z"
 
+rfc5422DateTimeFormatLax :: String
+rfc5422DateTimeFormatLax = "%a, %-d %b %Y %-H:%-M:%-S %z"
```

Comment:
To clarify, the `%-H:%-M:%-S` allows the h/m/s without leading zero, correct?

Reply:
yes

Reply:
I think `%-d` is correct, but the RFC 5322 (and its predecessors) are clear that H/M/S are all 2DIGIT i.e. elision of leading `0` is prohibited.  Did you encounter this in practice?

Reply:
I encountered `%-d` in practice and added the others out of Postel's Law.  Could do only `%-d` for now if that makes you more comfortable until we have an example -- though not sure what the harm is in accepting a date header we can easily know the meaning of.

Reply:
It's fine for now, just good to know you did not encounter the others in practice.  There are several changes that need to be made to date parsing that were not previously (e.g. day of week is options, seconds are optional, and other issues; see https://tools.ietf.org/html/rfc5322#section-3.3).  So I'll accept this change for now because the main behavioural change is a big improvement, and make further changes later.